src/test/java/uk/gov/pay/api/service/GetPaymentRefundsServiceTest.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.pay.api.service;
+
+import au.com.dius.pact.consumer.PactVerification;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.api.app.RestClientFactory;
+import uk.gov.pay.api.app.config.PublicApiConfig;
+import uk.gov.pay.api.app.config.RestClientConfig;
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.ledger.service.LedgerUriGenerator;
+import uk.gov.pay.api.model.RefundResponse;
+import uk.gov.pay.api.model.RefundsResponse;
+import uk.gov.pay.api.model.TokenPaymentType;
+import uk.gov.pay.commons.testing.pact.consumers.PactProviderRule;
+import uk.gov.pay.commons.testing.pact.consumers.Pacts;
+
+import javax.ws.rs.client.Client;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetPaymentRefundsServiceTest {
+
+    private static final String ACCOUNT_ID = "123456";
+    @Rule
+    public PactProviderRule ledgerRule = new PactProviderRule("ledger", this);
+    @Mock
+    private PublicApiConfig mockConfiguration;
+    @Mock
+    private ConnectorService mockConnectorService;
+    private GetPaymentRefundsService getPaymentRefundsService;
+
+    @Before
+    public void setUp() {
+        when(mockConfiguration.getLedgerUrl()).thenReturn(ledgerRule.getUrl());
+        when(mockConfiguration.getBaseUrl()).thenReturn("http://publicapi.test.localhost/");
+
+        PublicApiUriGenerator publicApiUriGenerator = new PublicApiUriGenerator(mockConfiguration);
+        LedgerUriGenerator ledgerUriGenerator = new LedgerUriGenerator(mockConfiguration);
+
+        Client client = RestClientFactory.buildClient(new RestClientConfig(false));
+        LedgerService ledgerService = new LedgerService(client, ledgerUriGenerator);
+
+        getPaymentRefundsService = new GetPaymentRefundsService(mockConnectorService, ledgerService, publicApiUriGenerator);
+    }
+
+    @Test
+    @PactVerification("ledger")
+    @Pacts(pacts = {"publicapi-ledger-get-payment-refunds"})
```

Comment:
I'm not convinced by the name (of the file).
Shouldn't it be more like `publicapi-ledger-get-transaction-refunds` or
`publicapi-ledger-get-transaction-transaction` (this one doesn't sound great either though).

Reply:
This naming makes it clear of what we are testing (and the action) from publicapi point of view. Ledger state is named `refund transactions for a transaction exist` which makes it relevant within ledger

We follow similar naming convention for all ledger and connector pacts